mlflow/transformers.py

```diff
@@ -0,0 +1,789 @@
+                       `transformers_model`. If not provided, an attempt will be made to fetch
+                       the card from the base pretrained model that is provided (or the one that is
+                       included within a provided `Pipeline`).
+    :param conda_env: A dictionary specifying a conda environment to be created for running the
+                      model. The environment can be specified as a YAML file, a string in YAML or
+                      JSON format, or a dictionary.
+    :param code_paths: A list of local filesystem paths to Python file dependencies (or directories
+                       containing file dependencies). These files are *prepended* to the system
+                       path when the model is loaded.
+    :param mlflow_model: An MLflow model object that specifies the flavor that this model is being
+                         added to.
+    :param signature: A Model Signature object that describes the input and output Schema of the
+                      model. The model signature can be inferred using `infer_signature` function
+                      of `mlflow.models.signature`.
+    :param input_example: An example of valid input that the model can accept. The example can be
+                          used as a hint of what data to feed the model. The given example will be
+                          converted to a `Pandas DataFrame` and then serialized to JSON using the
+                          `Pandas` split-oriented format. Bytes are base64-encoded.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+
+    :param kwargs: Optional additional configurations for transformers serialization.
+    :return: None
+    """
+    import transformers
+
+    _validate_transformers_model_dict(transformers_model)
+
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    path = pathlib.Path(path).absolute()
+
+    _validate_and_prepare_target_save_path(str(path))
+
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, str(path))
+
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, str(path))
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    resolved_task = _get_or_infer_task_type(transformers_model, task)
+
+    # build the pipeline object if a model is passed
+    if not isinstance(transformers_model, transformers.Pipeline):
+        built_pipeline = _build_pipeline_from_model_input(transformers_model, resolved_task)
```

Comment:
From the docstring:

```
                      .. Note:: If a processor is supplied with the saving a model, the
                                model will be unavailable for loading as a ``Pipeline``
```

If the model can't be loaded as a pipeline when saved with a processor, does this logic throw, or can we in fact correctly save the model as a pipeline and load it?

Reply:
I'm going to add a unit test to verify this

Reply:
added: `test_model_with_processor_does_not_function()` to verify that a processor type mangles the expected input types internally within the pipeline abstraction, making it so that passed-in data encodings cannot be subsequently processed properly with the next expected submodule object.